packages/deploy/src/stateTransform.ts

```diff
@@ -20,7 +21,25 @@ import {
 import { DeployError } from './deployError';
 import { Logger } from '@openfn/logger/dist';
 
+function getStateJobCredential(
+  specJobCredential: undefined | null | string,
+  stateCredentials: ProjectState['project_credentials']
+): undefined | null | string {
+  if (
```

Comment:
I would simply set the condition to
```
if (!stateCredentials[specJobCredential]) {
```

That's more idiomatic, concise, and safely covers what you're looking for